### PR TITLE
Migrate to pyproject.toml, remove setup.cfg, pyscaffold

### PR DESCRIPTION
**Pull request by andmat900**
### Applicable Issues
- https://github.com/eiffel-community/etos/issues/208

### Description of the Change
This change removed setup.cfg and requirements.txt, migrating the package to pyproject.toml

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:  Andrei Matveyeu, andrei.matveyeu@axis.com

### CODE REVIEW

File: src/etos_test_runner/lib/executor.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def _finished(self, test_name, result):
 
     def _call(
         self, cmd, shell=False, env=None, executable=None, output=None, wait_output=True
-    ):  # pylint:disable=too-many-arguments
+    ):  # pylint:disable=too-many-positional-arguments,too-many-arguments
```
Comment (t-persson) [inline src/etos_test_runner/lib/executor.py:252]: Do we really need both?

Comment (andmat900) [reply src/etos_test_runner/lib/executor.py:252]: Yes, both are needed.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+[build-system]
+requires = ["setuptools>=72", "wheel", "setuptools_scm[toml]>=8"]
+
+[project]
+name = "etos_test_runner"
+dynamic = ["version"]
+description = "ETOS Test Runner"
+authors = [{name = "Tobias Persson", email = "tobias.persson@axis.com"}]
```
Comment (fredjn) [inline pyproject.toml:8]: Do we want to put all the "blame" on @t-persson ? ;)

Comment (fredjn) [reply pyproject.toml:8]: It's kind of unclear what this field describes/conveys?

**Review by fredjn — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (andmat900) [reply pyproject.toml:8]: I guess this parameter must be present here with some value. Maintaining the list of all authors would be difficult in each repo's `pyproject.toml`. Sometimes there's also an author list in the documentation. 

If necessary, it probably would be better to generate this list from the Git history.

**Pull request merged**